From 736854ae19d9186a1ebe4d93eaeb5b5bc568f4c4 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Thu, 16 May 2024 11:09:02 +0200
Subject: [PATCH] ARM: dts: stm32: add pins muxing for mdf on stm32mp21

These pins are used for MDF on STM32MP215 Disco board.

Change-Id: I042f9d591948a605c6da12efc2393556d91b9607
Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/380183
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi | 30 +++++++++++++++++++
 1 file changed, 30 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi
@@ -23,6 +23,36 @@
 		};
 	};
 
+	mdf_cck0_pins_a: mdf-cck0-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('F', 3, AF5)>; /* MDF1_CCK0 */
+			slew-rate = <1>;
+			drive-push-pull;
+			bias-disable;
+		};
+	};
+
+	mdf_cck0_sleep_pins_a: mdf-cck0-sleep-0 {
+		pins {
+			pinmux = <STM32_PINMUX('F', 3, ANALOG)>; /* MDF1_CCK0 */
+		};
+	};
+
+	mdf_sdi3_pins_a: mdf-sdi3-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('B', 3, AF5)>; /* MDF1_SDI3 */
+			slew-rate = <1>;
+			drive-push-pull;
+			bias-disable;
+		};
+	};
+
+	mdf_sdi3_sleep_pins_a: mdf-sdi3-sleep-0 {
+		pins {
+			pinmux = <STM32_PINMUX('B', 3, ANALOG)>; /* MDF1_SDI3 */
+		};
+	};
+
 	sdmmc1_b4_pins_a: sdmmc1-b4-0 {
 		pins1 {
 			pinmux = <STM32_PINMUX('E', 4, AF10)>, /* SDMMC1_D0 */
